lasagne/layers/normalization.py

```diff
@@ -101,3 +115,233 @@ def get_output_for(self, input, **kwargs):
             scale += self.alpha * input_sqr[:, i:i+ch, :, :]
         scale = scale ** self.beta
         return input / scale
+
+
+class BatchNormLayer(Layer):
+    """
+    lasagne.layers.BatchNormLayer(incoming, axes='auto', epsilon=1e-4,
+    alpha=0.5, nonlinearity=None, mode='low_mem',
```

Comment:
Hmm, good question. When I did my first experiments, there was no other implementation yet. It would probably be good to use similar values as other libraries just in case people switch between them.

* Torch has 0.1: https://github.com/torch/nn/blob/eb13695/BatchNormalization.lua#L46
* Caffe has 0.01: https://github.com/BVLC/caffe/blob/39f69fbe520d90d30fc342e5d69153ec9534d2bc/src/caffe/proto/caffe.proto#L476
* TensorFlow: Can't find anything about running mean/variance: https://tensorflow.googlesource.com/tensorflow/+/master/tensorflow/core/kernels/batch_norm_op.cc
* Keras has 0.9, which means 0.1: https://github.com/fchollet/keras/blob/e839a4b/keras/layers/normalization.py#L19
* MXNet has 0.9, which probably means 0.1: http://mxnet.readthedocs.org/en/latest/python/symbol.html#mxnet.symbol.BatchNorm

I'll change it to 0.1 then.

We *could* also rename it to `momentum` and invert the meaning, defaulting it to 0.9, but I think that's a bit misleading: In `lasagne.updates.momentum`, velocities are updated as `x := momentum * x + new_x`, while here the formula would be `x := momentum * x + (1 - momentum) * new_x` (i.e., the new value is properly scaled here while it's not scaled in `lasagne.updates`).